### ISSUE TITLE
bytes object does not support item assignment

### ISSUE THREAD

**Issue by paulbaumgarten** [KitronikLtd/micropython-microbit-kitronik-klef-piano#2]
Hello

I'm using the example code on a Kitronik Klef Piano 5631 v1.0 using Mu editor 1.0.2 running Python 3.6.3

I'm getting a TypeError on the bytearray assignments. Specifically, line 74, 'bytes' object does not support item assignment.

```python
buff[0] = 2
```

While I've done quite a bit of Python programming, I haven't used the bytesobject or written my own interfaces for i2c before so any advice would be appreciated. Being the sample code I would have thought it should "just work", hence the issue report.

Many thanks

**Comment by kevinjwalters** [comment]
The problem here comes from reusing the `buff` variable and types. The inital type is a `bytearray` at line 31 

https://github.com/KitronikLtd/micropython-microbit-kitronik-klef-piano/blob/master/klef-piano.py#L30-L82

_but_ there's some subsequent code inside the `if` statement at line 41/43 which assigns the `i2c.read()` return value to `buff` _replacing_ the `bytearray`. The documentation for the microbit version (some care is needed here as the i2c interface changes depending on the board) of [i2c.read](https://microbit-micropython.readthedocs.io/en/latest/i2c.html#microbit.i2c.read) does not reveal the return type. Based on your observations it's likely to be [bytes](https://docs.python.org/3/library/stdtypes.html#bytes-objects) which is an immutable type in Python and MicroPython, hence assigment afterwards to the element(s) fails, as expected.

Checking on a board over webconsole confirms the return type, reading a value from an onboard peripherial (there are three variants of this based on board revision, btw) confirms this:

```
MicroPython v1.9.2-34-gd64154c73 on 2017-09-01; micro:bit v1.0.1 with nRF51822
Type "help()" for more information.
>>> i2c
<MicroBitI2C>
>>> buff = i2c.read(0x1d, 1, False)
>>> type(buff)
<class 'bytes'>
>>> buff[0]
255
>>> buff[0] = 254  # bytes is immutable...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'bytes' object does not support item assignment
```

Some general tidying up of the use of buffers would be useful here in addition to fixing the above.

It looks like @AlasdairAtKitronik has worked on this in past.

**Comment by AlasdairAtKitronik** [comment]
@paulbaumgarten @kevinjwalters Thanks for flagging this issue, I'll aim to have a look into it this week. You're right, the example should just work, and when it was uploaded, it all worked fine, and has been for several months - I'll need to see what's changed elsewhere.

**Comment by AlasdairAtKitronik** [comment]
Example program fixed. Thank you for your input @kevinjwalters.

**Comment by kevinjwalters** [comment]
@paulbaumgarten @AlasdairAtKitronik The program didn't work for me. There's a one line fix in #3 for this.
